### PR TITLE
Initial 1.18 commit

### PR DESCRIPTION
**Pull request by Stelk**
Just the minimum changes to get the code working on 1.18-pre5.  
version.properties may need to be excluded from the changeset to keep the mod version consistent with future releases.

I'll provide another PR once MC 1.18 is officially released on the 30th.

### CODE REVIEW

File: src/main/java/ru/bulldog/justmap/client/render/BufferedRenderer.java [hunk 1]
```diff
@@ -109,7 +109,7 @@ protected void render(MatrixStack matrices, double scale) {
 		RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
 		RenderSystem.setShader(GameRenderer::getPositionTexShader);
 		// yarn mapping missing for beginRead
```
Comment (magicus) [inline src/main/java/ru/bulldog/justmap/client/render/BufferedRenderer.java]: You can remove the comment now that it's not relevant

File: src/main/java/ru/bulldog/justmap/client/render/BufferedRenderer.java [hunk 2]
```diff
@@ -154,7 +154,7 @@ protected void render(MatrixStack matrices, double scale) {
 		matrices.multiply(Vec3f.POSITIVE_Z.getDegreesQuaternion(180.0F));
 		RenderSystem.setShader(GameRenderer::getPositionTexShader);
 		// yarn mapping missing: beginRead
```
Comment (magicus) [inline src/main/java/ru/bulldog/justmap/client/render/BufferedRenderer.java]: Same here

File: src/main/java/ru/bulldog/justmap/map/data/classic/ChunkDataManager.java [hunk 3]
```diff
@@ -1,21 +1,20 @@
 package ru.bulldog.justmap.map.data.classic;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (magicus) [inline src/main/java/ru/bulldog/justmap/map/data/classic/ChunkDataManager.java]: The code base contains no `*` imports. I think this is a code style that is good to keep.

File: src/main/java/ru/bulldog/justmap/map/data/classic/ChunkDataManager.java [hunk 4]
```diff
@@ -114,12 +113,14 @@ private WorldChunk callSaves(World world, ChunkPos chunkPos) {
...
-					DataUtil.getPersistentSupplier(), storage.getNbt(chunkPos));
+		try (VersionedChunkStorage storage = StorageUtil.getChunkStorage(serverWorld);) {
+			//guessing here that null might work for the optional parameter.  Hopefully there is a null check.
+			Optional<RegistryKey<Codec<? extends ChunkGenerator>>> opt = Optional.ofNullable(null);
+			NbtCompound chunkTag = storage.updateChunkNbt(serverWorld.getRegistryKey(), DataUtil.getPersistentSupplier(), storage.getNbt(chunkPos), opt);
 			if (chunkTag == null) return this.emptyChunk;
 			Chunk chunk = ChunkSerializer.deserialize(
-					serverWorld, serverWorld.getStructureManager(), serverWorld.getPointOfInterestStorage(), chunkPos, chunkTag);
+					serverWorld, serverWorld.getPointOfInterestStorage(), chunkPos, chunkTag);
+					// serverWorld, serverWorld.getStructureManager(), serverWorld.getPointOfInterestStorage(), chunkPos, chunkTag);
```
Comment (magicus) [inline src/main/java/ru/bulldog/justmap/map/data/classic/ChunkDataManager.java]: You can remove the commented out code

File: src/main/java/ru/bulldog/justmap/util/render/ExtendedFramebuffer.java [hunk 5]
```diff
@@ -321,7 +321,7 @@ public void endWrite() {
 
 	// yarn mapping missing for beginRead
```
Comment (magicus) [inline src/main/java/ru/bulldog/justmap/util/render/ExtendedFramebuffer.java]: Comment not relevant anymore

**Review by magicus — CHANGES_REQUESTED** [review]

**Comment by Bulldog83** [comment]
There are still conflicts here.

File: src/main/java/ru/bulldog/justmap/client/render/BufferedRenderer.java (hunk 1 above)
Comment (Stelk) [reply src/main/java/ru/bulldog/justmap/client/render/BufferedRenderer.java]: Removed

File: src/main/java/ru/bulldog/justmap/client/render/BufferedRenderer.java (hunk 2 above)
Comment (Stelk) [reply src/main/java/ru/bulldog/justmap/client/render/BufferedRenderer.java]: Removed

File: src/main/java/ru/bulldog/justmap/map/data/classic/ChunkDataManager.java (hunk 3 above)
Comment (Stelk) [reply src/main/java/ru/bulldog/justmap/map/data/classic/ChunkDataManager.java]: No problem.  Looks like Intellij wil automatically replace the imports with a wildcard once more than five from the same package are specified.  * has been removed.

File: src/main/java/ru/bulldog/justmap/map/data/classic/ChunkDataManager.java (hunk 4 above)
Comment (Stelk) [reply src/main/java/ru/bulldog/justmap/map/data/classic/ChunkDataManager.java]: Removed

File: src/main/java/ru/bulldog/justmap/util/render/ExtendedFramebuffer.java (hunk 5 above)
Comment (Stelk) [reply src/main/java/ru/bulldog/justmap/util/render/ExtendedFramebuffer.java]: Removed

**Comment by Stelk** [comment]
My bad.   In my fork 1.18 is a branch of 1.17 while I believe upstream branched 1.18 from master.  I'll rebase/test/and commit the changes.  I should have asked how the branching is managed beforehand.  Thanks for the feedback.

**Comment by Bulldog83** [comment]
@magicus works with 1.17 version, his PRs in the 'master' branch. I've started 1.17 branch before, but did't complete it.

**Pull request closed without merging**